data_structures/binary_tree/binary_tree_traversals.py

```diff
@@ -0,0 +1,161 @@
+class Node:
+    """
+    A Node has data variable and pointers toits left and right nodes.
+    """
+
+    def __init__(self, data):
+        self.left = None
+        self.right = None
+        self.data = data
+
+
+def preorder(root):
+    """
+    PreOrder traversal: visit root node then its left subtree followed by right subtree.

```

Comment:
```suggestiondef make_tree() -> Node:
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)
    root.left.right = Node(5)
    return root


def preorder(root):
    """
    PreOrder traversal: visit root node then its left subtree followed by right subtree.

    >>> preorder(make_tree())
    4 2 5 1 3
```
This adds a utility function to and a doctest.  You can run this test at home with
`python3 -m doctest -v binary_tree_traversals.py`

Please add doctests to the other traversal functions so we know that they work as expected.